inc/Engine/Optimization/GoogleFonts/Combine.php

```diff
@@ -76,7 +76,11 @@ public function optimize( $html ) {
 			return $html;
 		}
 
-		$html = str_replace( '</title>', '</title>' . $this->get_combine_tag(), $html );
+		$head_html = preg_match( '/<head[^>]*>(.*?)<\/head>/umsi', $html, $matches );
+		if ( $head_html ) {
+			$new_head = preg_replace( '/<\/title>/', '</title>' . $this->get_combine_tag(), $matches[0], 1 );
+			$html     = str_replace( $matches[0], $new_head, $html );
+		}
```

Comment:
@Tabrisrp Challenge accepted.

```suggestion
		$html = preg_replace( '@(<head[^>]*>.*<title[^>]*>.*</title>)(.*</head>)@isU', '$1' . $this->get_combine_tag() . '$2', $html, 1 );
```

[Test](https://regex101.com/r/MfusIp/1).